qiskit/qpy/__init__.py

```diff
@@ -126,6 +126,17 @@
 by ``num_circuits`` in the file header). There is no padding between the
 circuits in the data.
 
+.. _qpy_version_7:
+
+Version 7
+=========
+
+Version 7 is identical to version 6, except two new type keys are added to the INSTRUCTION_PARAM
```

Comment:
This block can go to 
https://github.com/Qiskit/qiskit-terra/blob/912ba0783420463b4adf7f53bee7cba04bd915f8/qiskit/qpy/__init__.py#L905-L919

maybe we need better document structure.

Reply:
My problem was that that block is under the "version 1" heading, and the new type keys aren't defined in version 1, so it didn't seem right to put it there.  For example, in version 4, we add two type keys to the same `INSTRUCTION_PARAMS` struct, but don't put them in version 1.

Reply:
Yeah, I think we need to rework the documentation. At the time it was trying to build off what each version's structs looked like so you'd incrementally build the formats off each other. I think we probably need to reorganize it so the current format is at the top and then we describe (with full examples what things looked like in older versions). I'll open an issue about this so we can track it as something in a follow up. For this we can probably leave the new addition in just the version 7 section and fix this in that followup.